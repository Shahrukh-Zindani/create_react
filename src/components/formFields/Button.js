import React from 'react'

const ButtonField = (props) => (
  <button>{props.children}</button>
)

export default ButtonField
