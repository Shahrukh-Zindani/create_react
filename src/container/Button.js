import React from 'react'

class ButtonField extends React.Component {
  static propTypes = {
    onClick: React.PropTypes.func
  }
  render() {
    return (
      <button onClick={this.props.onClick}> + </button>
    )
  }
}

export default ButtonField
