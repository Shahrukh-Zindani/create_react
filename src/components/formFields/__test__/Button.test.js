import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Button from '../Button'

// testing the dumb component
it('renders without crashing', () => {
  const wrapper = shallow(<Button>{'Test'}</Button>)
  const test = 'Test'
  expect(wrapper.contains(test)).toEqual(true)
})
it('matches snapshot', () => {
  const tree = renderer.create(
    <Button />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
