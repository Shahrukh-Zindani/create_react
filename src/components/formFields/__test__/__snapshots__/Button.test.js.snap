exports[`test matches snapshot 1`] = `<button />`;
